pytorch_lightning/strategies/single_device.py

```diff
@@ -46,7 +49,7 @@ def on_tpu(self) -> bool:
     def on_gpu(self) -> bool:
         return self.root_device.type == "cuda" and torch.cuda.is_available()
 
-    def reduce(self, tensor: Union[Any, torch.Tensor], *args: Any, **kwargs: Any) -> Union[Any, torch.Tensor]:
+    def reduce(self, tensor: Any | torch.Tensor, *args: Any, **kwargs: Any) -> Any | torch.Tensor:
```

Comment:
noob question: how is it different from `-> Any` (without `torch.Tensor`)?

Reply:
I believe @justusschock advocated for this in the past.

Reply:
To elaborate myself, this is useful when it's `Any` but you want to include the information that it can also be those union types.

For example, for this snippet

```python
from typing import Union, Any

x: Union[Any, int, str]

reveal_type(x)
```

The revealed type is

```python
main.py:5: note: Revealed type is "Union[Any, builtins.int, builtins.str]"
```

So mypy does not throw away this extra information

I guess this is also why `Optional[Any] == Union[Any, None]` is preferred over just `Any`